sdk/agrifood/azure-agrifood-farming/samples/async/sample_attachments_async.py

```diff
@@ -0,0 +1,194 @@
+    await client.farmers.create_or_update(
+        farmer_id=farmer_id,
+        farmer=Farmer()
+    )
+    print("Done!")
+
+    # Ensure farm exists, create if necessary.
+    print(f"Create/updating farm with id {farm_id}...", end=" ", flush=True)
+    await client.farms.create_or_update(
+        farmer_id=farmer_id,
+        farm_id=farm_id,
+        farm=Farmer()
+    )
+    print("Done!")
+
+    # Create attachment on farmer
+    try:
+        print(f"Checking if attachment with id {attachment_on_farmer_id} already exists "
+            f"on farmer with id {farmer_id}...", end=" ", flush=True)
+        await client.attachments.get(
+            farmer_id=farmer_id,
+            attachment_id=attachment_on_farmer_id
+        )
+        print("Attachment already exists. Not updating file.")
+
+    except ResourceNotFoundError:
+        print("Attachment doesn't exist")
+        print("Creating attachment...", end=" ", flush=True)
+
+        # Open file with buffering set to 0, to get a IO object.
+        file_to_attach_on_farmer = open(
+            attachment_on_farmer_file_path,
+            "rb",
+            buffering=0)
+
+        await client.attachments.create_or_update(
+            farmer_id=farmer_id,
+            attachment_id=attachment_on_farmer_id,
+            resource_id= farmer_id,
+            resource_type="Farmer",
+            file=file_to_attach_on_farmer)
+
+        print("Done!")
+        
+    # Create attachment with farm
+    try:
+        print(f"Checking if attachment with id {attachment_on_farm_id} already exists " + 
+            f"on farm with id {farm_id}...", end=" ", flush=True)
+        await client.attachments.get(
+            farmer_id=farmer_id,
+            attachment_id=attachment_on_farm_id
+        )
+        print("Attachment already exists. Not updating file.")
+
+    except ResourceNotFoundError:
+        print("Attachment doesn't exist")
+        print("Creating attachment...", end=" ", flush=True)
+
+        # Open file with buffering set to 0, to get a IO object.
+        file_to_attach_on_farm = open(
+            attachment_on_farm_file_path,
+            "rb",
+            buffering=0)
+
+        await client.attachments.create_or_update(
+            farmer_id=farmer_id,
+            attachment_id=attachment_on_farm_id,
+            resource_id= farm_id,
+            resource_type="Farm",
+            file=file_to_attach_on_farm)
+
+        print("Done!")
+    
+    print("Proceeding to download all attachments on the farmer. " +
+        "Press enter to continue...")
+    input()
+
+
+    print("Getting a list of all attachments " +
+        f"on the farmer with id {farmer_id}...", end=" ", flush=True)
+    farmer_attachments = client.attachments.list_by_farmer_id(
+        farmer_id=farmer_id,
+    )
+    print("Done!")
+
+    # Using a semaphore to limit the number of concurrent downloads.
+    semaphore =  asyncio.Semaphore(2)
```

Comment:
There could be an unbounded number of files that might be downloaded, we want to stay clean even in those cases. In some internal trials, we saw that happen. Also, just to make it explicit in case somebody decides to blindly copy paste this.

Reply:
Confirmed with folks on my end, we wanna keep this.